### PR TITLE
Request keeper test 💯

### LINKED ISSUE #1712: Request keeper test
There are 3 functions need to be tested

- GetRandomValidators
- SaveResult
- ProcessExpiredRequests

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1712

### CODE REVIEW

File: chain/x/oracle/keeper/request_test.go [hunk 1]
```diff
@@ -135,3 +135,124 @@ func TestGetRandomValidatorsTooBigSize(t *testing.T) {
...
+	require.Equal(t, id, res.RequestID)
+	require.Equal(t, int64(0), res.AnsCount)
+	require.Equal(t, requestTime, res.RequestTime)
+	require.Equal(t, resolveStatus, res.ResolveStatus)
+	require.Equal(t, int64(1581589090), res.ResolveTime)
+	require.Equal(t, result, res.Result)
+}
+
+func TestProcessExpiredRequests(t *testing.T) {
+	_, ctx, k := createTestInput()
```
Comment (sorawit) [inline chain/x/oracle/keeper/request_test.go:164]: Please comment explain the test situation.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/keeper/request_test.go [hunk 2]
```diff
@@ -135,3 +135,124 @@ func TestGetRandomValidatorsTooBigSize(t *testing.T) {
...
+	require.Error(t, err)
+	_, err = k.GetRequest(ctx, id2)
+	require.Error(t, err)
+	rq3, err = k.GetRequest(ctx, id3)
+	require.NoError(t, err)
+	require.Equal(t, rq3, request3)
+}
+
+// TODO: revisit this test
+func TestProcessExpiredRequestsNoRequestInStore(t *testing.T) {
```
Comment (sorawit) [inline chain/x/oracle/keeper/request_test.go:236]: Please iterate through store and check if the number of elements are the same

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/oracle/keeper/request_test.go [hunk 3]
```diff
@@ -135,3 +135,115 @@ func TestGetRandomValidatorsTooBigSize(t *testing.T) {
...
+
+	rep := types.NewReport(
+		Validator1.ValAddress, []types.RawReport{
+			types.NewRawReport(1, 1, []byte("data1/1")),
+			types.NewRawReport(2, 0, []byte("data2/1")),
+		},
+	)
+
+	k.AddReport(ctx, id1, rep)
+	k.AddReport(ctx, id2, rep)
```
Comment (taobun) [inline chain/x/oracle/keeper/request_test.go:209]: Test expired request

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/keeper/request_test.go [hunk 4]
```diff
@@ -135,3 +135,115 @@ func TestGetRandomValidatorsTooBigSize(t *testing.T) {
...
+	rawRequestID := []types.ExternalID{1, 2, 3}
+	resolveStatus := types.ResolveStatus_Success
+	result := []byte("beeb")
+	id := k.AddRequest(ctx, types.NewRequest(
+		oracleScriptID, calldata, vals, minCount, requestHeight,
+		requestTime, clientID, &ibcInfo, rawRequestID))
+
+	res := k.SaveResult(ctx, id, resolveStatus, result)
+
+	require.Equal(t, clientID, res.ClientID)
```
Comment (taobun) [inline chain/x/oracle/keeper/request_test.go]: Combine in 1 packet.

File: chain/x/oracle/keeper/request_test.go [hunk 5]
```diff
@@ -135,3 +135,115 @@ func TestGetRandomValidatorsTooBigSize(t *testing.T) {
...
+
+	reports1 = k.GetReports(ctx, id1)
+	reports2 = k.GetReports(ctx, id2)
+
+	info := k.GetValidatorReportInfoWithDefault(ctx, Validator1.ValAddress)
+	require.Equal(t, types.NewValidatorReportInfo(Validator1.ValAddress, 0), info)
+
+	require.Equal(t, []types.Report(nil), reports1) // report1 was removed because it already expired
+	require.Equal(t, []types.Report{rep}, reports2)
+
```
Comment (taobun) [inline chain/x/oracle/keeper/request_test.go:221]: Get result of expired request it must have result in store.

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/keeper/request_test.go [hunk 6]
```diff
@@ -135,3 +135,116 @@ func TestGetRandomValidatorsTooBigSize(t *testing.T) {
...
+	res, err := k.GetResult(ctx, id1)
```
Comment (taobun) [inline chain/x/oracle/keeper/request_test.go]: Can add comment to explain what you tested

**Review by taobun — CHANGES_REQUESTED** [review]

**Comment by Benzbeeb** [comment]
I’m closing this pull request because of outdated code. Now we have go-owasm.([link](https://github.com/bandprotocol/bandchain/pull/1858))

**Pull request closed without merging**